Spanner/tests/System/GeneratedAdminEmulatorTest.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Copyright 2024 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Spanner\Tests\System;
+
+use Google\Cloud\Spanner\Admin\Database\V1\Client\DatabaseAdminClient;
+use Google\Cloud\Spanner\Admin\Database\V1\CreateDatabaseRequest;
+use Google\Cloud\Spanner\Admin\Database\V1\GetDatabaseRequest;
+use Google\Cloud\Spanner\Admin\Instance\V1\Client\InstanceAdminClient;
+use Google\Cloud\Spanner\Admin\Instance\V1\CreateInstanceRequest;
+use Google\Cloud\Spanner\Admin\Instance\V1\GetInstanceRequest;
+use Google\Cloud\Spanner\Admin\Instance\V1\Instance;
+
+/**
+ * @group spanner
+ */
+class GeneratedAdminEmulatorTest extends SpannerTestCase
+{
+    private static $projectId;
+
+    public static function setUpBeforeClass(): void
+    {
+        self::$projectId = 'emulator-project';
+    }
+
+    public function testAdminClientEmulatorSupport()
+    {
+        if (!(bool) getenv("SPANNER_EMULATOR_HOST")) {
```

Comment:
Do we need to explicitly typecast here?

Reply:
A better check would be `!empty`

Reply:
Changed

Reply:
`getenv` returns false when the env variable doesn't exists. So we can actually remove the typecasting completely. Could you try this?

ref: https://www.php.net/manual/en/function.getenv.php

Reply:
I wanted to check for the empty string codition as well

Reply:
Empty string is evaluated as false by default in PHP. 

```
<?php
$x = "";
if ($x) echo "Empty string true\n";
else echo "Empty string false\n";

// Prints "Empty string false"
```

Reply:
Cool, changed